packages/devtools/lib/src/memory/memory.dart

```diff
@@ -321,26 +332,129 @@ class MemoryScreen extends Screen with SetStateMixin {
       ..element.clazz('memory-table');
 
     try {
-      final List<InstanceSummary> instanceRows = await memoryController
-          .getInstances(row.classRef.id, row.instancesCurrent);
+      final List<InstanceSummary> instanceRows =
+          await memoryController.getInstances(
+              row.classRef.id, row.classRef.name, row.instancesCurrent);
 
       table.addColumn(new MemoryColumnSimple<InstanceSummary>(
           '${instanceRows.length} Instances of ${row.classRef.name}',
           (InstanceSummary row) => row.objectRef));
 
       table.setRows(instanceRows);
     } catch (e) {
-      framework.toast('Problem fetching instances $e', title: 'Error');
+      framework.toast(
+        'Problem fetching instances of ${row.classRef.name} $e',
+        title: 'Error',
+      );
     }
 
     table.onSelect.listen((InstanceSummary row) async {
       ga.select(ga.memory, ga.inspectInstance);
-      // TODO(terry): Handle clicking on an instance bring up variable inspector
+
+      try {
+        final Instance instance =
+            await memoryController.getObject(row.objectRef);
+
+        if (tableContainer.element.children.length > 2) {
+          tableContainer.element.children.removeLast();
+        }
+
+        tableContainer.add(_createInstanceView(
+          row.objectRef,
+          row.className,
+          instance.fields,
+        ));
+
+        tableContainer.element.scrollTo(<String, dynamic>{
+          'left': tableContainer.element.scrollWidth,
+          'top': 0,
+          'behavior': 'smooth',
+        });
+
+        // Allow inspection of the memory object.
+        memoryDataView.showFields(instance.fields);
+      } catch (e) {
+        framework.toast(
+          'Problem fetching instance ${row.objectRef} $e',
```

Comment:
maybe just 
'Unable to fetch instance ${row.objectRef}'
just because this probably doesn't mean a serious error as discussed above.

Reply:
Done.